### DISCUSSION TITLE
How to terminate a running ffmpegio stream forcibly?

### DISCUSSION THREAD

**Discussion by suurj** [python-ffmpegio/python-ffmpegio#7] [Q&A] [answered]
Hello!

Is there a way to terminate an ongoing ffmpegio stream like the one below, if it runs in its own thread:
```
with ffmpegio.open('http://stream.url', 'ra', blocksize=5000, ac=2, sample_fmt='flt', reconnect_at_eof=1,reconnect_streamed=1) as stream:
    for block in stream:
         process(block)
```

One possibility could be to kill the ffmpeg by looking at its PPID, which is of course the PID of the main Python process the code is run from. But is there a more elegant way? I need to terminate the ffmpeg process and the thread in case they hang, but I have not found any robust method for that.

**Comment by tikuma-lsuhsc** [comment, marked as answer]
@suurj - the code snippet will terminate the FFmpeg process if it's not dead when Python process exits the context. If FFmpeg process has not been terminated (i.e., all frames were read) it calls `Popen.kill()` on the FFmpeg process. If it is not behaving as such, I need to investigate into in (need more info).

You can inspect how the stream closes here:

https://github.com/python-ffmpegio/python-ffmpegio/blob/4fd16e1b914d0be18154caf1b06f7d046bed55b2/src/ffmpegio/streams/SimpleStreams.py#L91

**Reply by suurj** [reply]
The close() method seems to work, thanks. I am sure I tested it before, but I guess I just messed up with something.

**Reply by tikuma-lsuhsc** [reply]
I'm surprised that you need to call it explicitly though. when you're code gets out of `with` statement, it should call it automatically
